sklearn/utils/estimator_checks.py

```diff
@@ -2165,6 +2173,21 @@ def check_parameters_default_constructible(name, Estimator):
                     estimator = Estimator(Ridge())
                 else:
                     estimator = Estimator(LinearDiscriminantAnalysis())
+            elif "estimators" in required_parameters:
```

Comment:
It seems that this whole estimator initialization is duplicated between `check_parameters_default_constructible` and `_tested_estimators`. Might be worth considering a unifying helper?

Reply:
I agree but I would do that in another PR or go for a better solution as proposed by @jnothman.